src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarFile.cs

```diff
@@ -377,11 +374,9 @@ private static string GetEntryNameForFileSystemInfo(FileSystemInfo file, int bas
             return ArchivingUtils.EntryFromPath(file.FullName, basePathLength, entryNameLength, ref entryNameBuffer, appendPathSeparator: isDirectory);
         }
 
-        // Constructs a PaxTarEntry for a base directory entry when creating an archive.
-        private static PaxTarEntry GetEntryForBaseDirectory(string name, ref char[] entryNameBuffer)
+        private static string GetEntryNameForBaseDirectory(string name, ref char[] entryNameBuffer)
```

Comment:
Isn't changing this from `PaxTarEntry` to `string` requiring the calling code to now inspect the file system more, since it needs to call stat on the `di.FullName` in order to construct the TarEntry?

Reply:
We need to call `stat` to get the permissions for the `TarEntry`.